### PR TITLE
*: refine the table.Table interface

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Better code!

> The bigger the interface, the weaker the abstraction -- Rob Pike

### What is changed and how it works?

Remove unnecessary methods.

Why there are so many allocator related methods in the `table.Table` interface?

```
type Table interface {
        ...
	AllocHandle(ctx sessionctx.Context) (int64, error)
	AllocHandleIDs(ctx sessionctx.Context, n uint64) (int64, int64, error)	
	Allocator(ctx sessionctx.Context, allocatorType autoid.AllocatorType) autoid.Allocator	
	AllAllocators(ctx sessionctx.Context) autoid.Allocators
}
```

### Check List 

Tests 

 - No code

Code changes

 - Has exported function/method change
 - Has interface methods change

Side effects

 - Code backward compatibility

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @tangenta @lysu

File: table/table.go [hunk 1]
```diff
@@ -191,14 +191,8 @@ type Table interface {
...
-	// AllocHandle allocates a handle for a new row.
-	AllocHandle(ctx sessionctx.Context) (int64, error)
-
-	// AllocHandleIDs allocates multiple handle for rows.
-	AllocHandleIDs(ctx sessionctx.Context, n uint64) (int64, int64, error)
-
-	// Allocator returns Allocator.
-	Allocator(ctx sessionctx.Context, allocatorType autoid.AllocatorType) autoid.Allocator
+	// Use AllAllocators(ctx).Get(allocatorType) instead
+	// Allocator(ctx sessionctx.Context, allocatorType autoid.AllocatorType) autoid.Allocator
```
Comment (coocood) [inline table/table.go]: Remove it?

File: table/table.go [hunk 2]
```diff
@@ -191,14 +191,8 @@ type Table interface {
...
 
 	// AllAllocators returns all allocators.
 	AllAllocators(ctx sessionctx.Context) autoid.Allocators
```
Comment (coocood) [inline table/table.go]: Rename it to `Allocators`?

**Comment by coocood** [comment]
LGTM

File: table/table.go (hunk 2 above)
Comment (tiancaiamao) [reply table/table.go]: Done

File: table/table.go (hunk 1 above)
Comment (tiancaiamao) [reply table/table.go]: Done

**Pull request merged**